frontend/javascripts/oxalis/controller/td_controller.js

```diff
@@ -74,6 +107,19 @@ class TDController extends React.PureComponent<Props> {
     this.initMouse();
   }
 
+  componentDidUpdate(prevProps: Props, prevState: State) {
+    if (
```

Comment:
The core problem is that `setTargetAndFixPosition` is not called when a new active node is selected via the 3 viewport. This function updates the rotation center of the viewport.
Thus to fix the #4472 :bug: we trigger the call manually whenever it is necessary.

Another solution would be to call this method whenever the mouse inside the 3d viewport is moved. This kind of fix would have been fast but also creates a lot of unnecessary calls to `setTargetAndFixPosition`.
But @philippotto and I decided to take the first/current solution as I already had implemented it.

Reply:
Ok thanks for elaborating, I think I get it now :)

However, to my current understanding the derived state is not needed and I'd like to try to get rid of it to simplify the code. (Also if you're interested, this article by the React team is a good read: https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html)
Maybe I'm overlooking something, but can't you simply change the condition in the next lines to something like:
```
if (
  maybeGetActiveNodeFromProps(this.props) !== maybeGetActiveNodeFromProps(prevProps) &&
  maybeGetActiveNodeFromProps(this.props) !== INVALID_ACTIVE_NODE_ID
) { ... }
```

and remove the state, constructor and getDerivedStateFromProps methods?
As I see it, this may result in an additional call to `setTargetAndFixPosition` if a group is activated in between nodes, but that's fine I would say.
What do you think? :)

Reply:
> What do you think? :)

I think this is a very good suggestion. As you wrote: The state, constructor, and getDerivedStateFromProps are not necessary. Thank you for such advanced suggestions and the link to the blog :+1: